x-pack/legacy/plugins/spaces/public/views/management/edit_space/customize_space/customize_space_avatar.tsx

```diff
@@ -73,11 +104,52 @@ class CustomizeSpaceAvatarUI extends Component<Props, State> {
               isInvalid={isInvalidSpaceColor}
             />
           </EuiFormRow>
+          {this.filePickerOrImage()}
```

Comment:
This view can be a tad confusing to reason about now. If an image is specified, then the initials and color have no impact on the space avatar, so users could become confused.

![image](https://user-images.githubusercontent.com/3493255/64551372-e19bf300-d302-11e9-8b9d-480d04bbdc7c.png)


What do you think about some sort of tabbed approach, where users choose to either specify initials and color, _or_ an avatar image? @elastic/kibana-design might have some thoughts on this too.

Reply:
Tabs are going to be tricky in this space. I think a common pattern here is to just use a filepicker input (specifically `EuiFilePicker` with the `compressed` prop) with a 'Custom image' label.

Upon uploading an image, the Initials and Color inputs become disabled and a link, trash can icon button appears to remove the custom image (thus re-enabling the Initials and Color).

Let me know if you need a mockup.

Reply:
Mockup based upon my comment above:

<img width="1462" alt="Screenshot 2019-09-09 13 52 15" src="https://user-images.githubusercontent.com/446285/64558235-0b581880-d309-11e9-80ba-7e27c5549a0a.png">

Reply:
I think the disabling of the previous controls works. I just had one extra suggestion once a user's has uploaded an image.

<img src="https://d.pr/free/i/XKx04f+" />

Reply:
I implemented the suggested mock-up. Just one difference: the tooltip above the trash icon is a "title" and not a popout. Let me know if this is ok.